### ISSUE TITLE
Error registering Keras serializable loss

### ISSUE THREAD

**Issue by artemmavrin** [tensorflow/tensorflow#36259]
**System information**
- Have I written custom code): yes
- TensorFlow installed from: binary
- TensorFlow version: 2.1
- Python version: 3.7

**Describe the current behavior**

Using the `tf.keras.utils.register_keras_serializable` decorator does not allow correct serialization/restoration of a custom `tf.keras.losses.Loss` subclass without using a `custom_objects` argument.

**Describe the expected behavior**

Using `tf.keras.utils.register_keras_serializable` allows serialization/restoration of custom losses without using a `custom_objects` argument.

**Code to reproduce the issue**

```python
import tensorflow as tf


@tf.keras.utils.register_keras_serializable()
class CustomLoss(tf.keras.losses.MeanSquaredError):
    pass


model = tf.keras.Sequential([tf.keras.layers.Dense(units=1, input_shape=(1,))])
model.compile(optimizer='sgd', loss=CustomLoss())
model.save('model')

# ValueError: Unknown loss function: CustomLoss
tf.keras.models.load_model('model', compile=True)
```

**Other info / logs**
Relevant traceback:

https://github.com/tensorflow/tensorflow/blob/fd242bad45bd9778d61ad116001fd4b191e51c30/tensorflow/python/keras/saving/save.py#L190

https://github.com/tensorflow/tensorflow/blob/fd242bad45bd9778d61ad116001fd4b191e51c30/tensorflow/python/keras/saving/saved_model/load.py#L114-L115

https://github.com/tensorflow/tensorflow/blob/fd242bad45bd9778d61ad116001fd4b191e51c30/tensorflow/python/keras/saving/saving_utils.py#L259

https://github.com/tensorflow/tensorflow/blob/fd242bad45bd9778d61ad116001fd4b191e51c30/tensorflow/python/keras/losses.py#L1301-L1305

https://github.com/tensorflow/tensorflow/blob/fd242bad45bd9778d61ad116001fd4b191e51c30/tensorflow/python/keras/utils/generic_utils.py#L361-L362

https://github.com/tensorflow/tensorflow/blob/fd242bad45bd9778d61ad116001fd4b191e51c30/tensorflow/python/keras/utils/generic_utils.py#L321

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.1.0-rc2, 2.2.0-dev20200128 and was able to reproduce the issue.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/55d11b811078c362ff6e49379da72353/untitled597.ipynb).Thanks!

**Comment by howl-anderson** [comment]
I found this bug too and fixed in #37018
